book/accessibility.md

```diff
@@ -2095,103 +2102,52 @@ knowing that it was implemented by a browser is a good sign that it's not
-
-    <div role=textbox>contents</div>
-
-[button-role]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
-
-Its role in the accessibility tree now becomes equivalent to a `<button>`
-element. The first text child is also reused as the label of the button, and
-all other descendants of the element become presentational. However, all other
-functionality of a `<button>` *does not occur by default*. In particular:
-
- * The elent is not by defafult focusable.
- * Visual styling is unchanged.
- * Event handlers for form submission, such as the `<enter>` key or mouse click,
-    are not added.
-
-That means that the web application---not the browser---is now responsible for
-implementing all of this correctly. And if the application doesn't do it, the
-user is left confused and sad, because the screen reader will claim the element
-is a button but it doesn't seem to work. That's why it's better for a web
-application author to simply use `<button>` elements---it's all too easy to
-accidentally forget to implement something important for those users.
-
-But the `button` role nevertheless exists, so that the web application doesn't
-lose accessibility when the page uses custom widgets for one reason or
-another.^[One common reason is a web app that was originally
-built without much attention to accessibility, but needs to be retrofitted.]
-Likewise, there is a `textbox` role that makes an element behave like
-an `<input>` element.^[Text boxes are even harder for the developer to
-implement, since support is needed for features such as editing partially
-written text and supporting more than one language.] There are in total a
-large number of [defined roles][aria-roles-list].
-
-[aria-roles-list]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques
-
-Instead of implementing those not-so-useful `button` and `textbox` roles, let's
-add support for the `alert` role, which *is* quite useful. This role causes the
-text contents of the element to be immediately announced to the user. That's
-super useful, because it allows the web application to tell the user in words
-what might otherwise have been explained in pictures.
+Live Regions
+============
 
-Now that roles and element tags need not be the same, we need to redefine
-`announce_text` to look only at the computed role, and not the element tag:
+Scripts do not interact directly with the accessibility tree,[^aom]
+much like they do not interact directly with the display list.
+However, sometimes scripts need to inform the screen reader about
+*why* they're making certain changes to the page to give screen-reader
+users a better experience. The most common example is an alert[^toast]
+telling you that some action you just did failed. Typically, these are
+inserted at some fixed place in the document, and thus in the
+accessibility tree; there's little chance a screen-reader user would
```

Comment:
The second half of this sentence is confusing. I think you're trying to say that you can't just rely on the regular document-reading order for alerts, but it's not clear enough.